sourceinfo/sourceinfo.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sourceinfo
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/bufbuild/protoschema-plugins/internal/protoschema/plugin/pluginsourceinfo"
+	"github.com/jhump/protoreflect/desc/sourceinfo"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protodesc"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+var (
+	// GlobalFiles is a replacement for protoregistry.GlobalFiles that includes
+	// all registered source info.
+	GlobalFiles protodesc.Resolver = sourceinfo.GlobalFiles
```

Comment:
Global variables are banned - Make this API not use global variables

Reply:
This was not resolved - you just changed this from a global variable to a static function that wraps a global variable.

Reply:
This is how the sourceinfo package exposes the functionality. Are you suggesting that package can't be used? Or do you have another solution in mind?